shotover/src/transforms/tee.rs

```diff
@@ -97,6 +117,7 @@ pub struct Tee {
     pub behavior: ConsistencyBehavior,
     pub timeout_micros: Option<u64>,
     dropped_messages: Counter,
+    switched_chain: Option<Arc<Mutex<bool>>>,
```

Comment:
This should be an `Arc<AtomicBool>` otherwise we'll have lock contention when there are multiple active connections at the same time.

Reply:
Its not clear from the name + type what `switched_chain` does.

Maybe we should use an atomic enum like this?
```rust
Tee {
    result_source: ResultSource
    ...
}

enum ResultSource {
     RegularChain,
     TeeChain,
}
```